design/proposals/encoded_token_response.md

```diff
@@ -0,0 +1,165 @@
+Otherwise, returns json access token as `application/json`.
+
+```ruby
+def authenticate
+    # get the authn access token
+    content_type = (request.headers["Accept"] == "text/plain" && request.headers["Accept-Encoding"] == "base64") ? :text : :json
+    # encode Base64 if needed
+    render content_type => auth_token
+```
+### Backwards compatibility
+Default behaviour: Only if the `Accept` is set to `text/plain` the response will be encoded.
+
+### Notes
+* Default behaviour causes that all other `Accept` content-types (and their encodings) are ignored (By requesting 
+`../authenticate` with `Accept: "xml", Accept-Encoding: "gzip"` will return decoded `application/json`).
+
+## Solution - URL Patten
+According to [Rails Routing Guides](https://guides.rubyonrails.org/routing.html#route-globbing-and-wildcard-segments)  we can use wildcards to suggests the requested `Content_type`.
+
+By requesting `../authenticate.encoded` the output will be encoded access token.
+
+### Design
+When `../authenticate` request the response will be encoded **only** if the url suffix (wildcard) is `.encoded`.
+Otherwise, returns json access token as `application/json`
+
+```ruby
+def authenticate
+    # get the authn access token
+    # extract the url suffix
+    content_type = ( params[:format] == "encoded") ? :text : :json
+    # encode Base64 if needed
+    render content_type => auth_token
+```
+### Backwards compatibility
+By adding `format: true` to the endpoint, it makes the wildcard **mandatory**. Which means that previous `../authenticate`
+request **will fail** unless we take care of non-format requests (can be done by adding a route without format or `respond_to` and `respond_with` keywords).
+
+## Omitted Solutions
+* **Optional Parameters** - By passing an optional parameter (`../authenticate?encoded=true` e.g.) which requires 
+an encoded response. This solution is not popular in Conjur because the server doesn't use url optional parameters.  
+
+* **New Endpoint** - By requesting to a new endpoint (`../authenticate/encoded` e.g.) the response will be encoded. This 
+solution is a redundant overhead which can (and should) be avoided. 
+
+## Preferred Solution
+Even though, rails routing guidelines suggests that it supports response's content type based on wildcard suffix,
+I prefer the `Accept` HTTP header solution. This solution is is more accurate from content negotiation point of view.
+The suffix based solution is less complicated but it makes the request seems like it ask for a file (which ends with a 
+certain suffix) rather than a dynamic content type handling. The `Accept` method is more like a contract between the client 
+and server - in our case the client accepts `text/plain` response causes the server to send an encoded token. 
+
+## Security
+TBD
+
+## Test Plan
+
+1. Tests should be on DAP and Conjur (HTTPS vs HTTP).
```

Comment:
should we test this for each authenticator or is it enough to test it with `authn`?

cc: @eladkug